extension-manifest-v3/src/background/telemetry.js

```diff
@@ -0,0 +1,53 @@
+import { Metrics as Telemetry } from '@ghostery/libs';
+import { store } from 'hybrids';
+
+import Options from '/store/options.js';
+import Metrics from '/store/metrics.js';
+import UTMs from '/store/utms.js';
+
+const log = console.log.bind(console, '[telemetry]');
+
+async function recordUTMs(telemetry, JUST_INSTALLED) {
+  if (JUST_INSTALLED) {
+    const { utm_source, utm_campaign } = await telemetry.detectUTMs();
+    await store.set(utms, { utm_source, utm_campaign });
+    return;
+  }
+
+  const utms = await store.resolve(store.get(UTMs));
+
+  telemetry.setUTMs(utms);
+}
+
+(async () => {
+  const options = await store.resolve(store.get(Options));
+  const metrics = await store.resolve(store.get(Metrics));
+
+  const telemetry = new Telemetry({
+    METRICS_BASE_URL: 'https://d.ghostery.com',
+    EXTENSION_VERSION: chrome.runtime.getManifest().version,
+    conf: options,
+    log,
+    // object copy as hybrics arrays are not extensible
+    storage: JSON.parse(JSON.stringify(metrics)),
```

Comment:
array properties of the model objects are immutable, thus I had to clone the object here. @smalluban how this should be used?

Reply:
By the design, the model structure is frozen, and cannot be changed after initialization. It is not a bad thing, as you discovered a weak spot of the `Telemetry` class - it should not mutate arguments passed to it :)
From the store perspective, the feature helps to make changes to the model by mistake thinking that it will update the model.